040-SDK/010-TypeScript/010-overview.mdx

```diff
@@ -101,3 +101,9 @@ export const MyComponent: FC<MyComponentProps> = ({ records }) => {
   );
 };
 ```
+
+## Limitations
+
+Type recursion is limited to 3 levels in the TypeScript SDK, because each nested level adds computational complexity to the type checker and the LSP.
```

Comment:
```suggestion
Type recursion is limited to 3 levels in the TypeScript SDK, because each nested level adds computational complexity to the type checker and the Language Server Protocol (LSP).
```